test/unit/models/classes/dataBinding/GenerisInstanceDataBinderTest.php

```diff
@@ -0,0 +1,717 @@
+    private $classType2;
+
+    /** @var core_kernel_classes_Property|MockObject */
+    private $property1;
+
+    /** @var core_kernel_classes_Property|MockObject */
+    private $property2;
+
+    /** @var EventManager|MockObject */
+    private $eventManagerMock;
+
+    /** @var core_kernel_classes_ContainerCollection|MockObject */
+    private $emptyCollectionMock;
+
+    /** @var core_kernel_classes_ContainerCollection|MockObject */
+    private $nonEmptyCollectionMock;
+
+    public function setUp(): void
+    {
+        $this->eventManagerMock = $this->createMock(EventManager::class);
+
+        $this->classType1 = $this->createMock(core_kernel_classes_Class::class);
+        $this->classType1
+            ->method('getUri')
+            ->willReturn(self::URI_TYPE_1);
+
+        $this->classType2 = $this->createMock(core_kernel_classes_Class::class);
+        $this->classType2
+            ->method('getUri')
+            ->willReturn(self::URI_TYPE_2);
+
+        $this->target = $this->createMock(
+            core_kernel_classes_Resource::class
+        );
+
+        $this->target
+            ->method('getClass')
+            ->willReturnMap([
+                [self::URI_TYPE_1, $this->classType1],
+                [self::URI_TYPE_2, $this->classType2],
+            ]);
+
+        $this->property1 = $this->createMock(core_kernel_classes_Property::class);
+        $this->property1
+            ->method('getUri')
+            ->willReturn(self::URI_PROPERTY_1);
+
+        $this->property2 = $this->createMock(core_kernel_classes_Property::class);
+        $this->property2
+            ->method('getUri')
+            ->willReturn(self::URI_PROPERTY_2);
+
+        $this->target
+            ->method('getUri')
+            ->willReturn('http://test/resource');
+
+        $this->nonEmptyCollectionMock = $this->createMock(
+            core_kernel_classes_ContainerCollection::class
+        );
+
+        $this->nonEmptyCollectionMock
+            ->method('count')
+            ->willReturn(1);
+
+        $this->emptyCollectionMock = $this->createMock(
+            core_kernel_classes_ContainerCollection::class
+        );
+
+        $this->emptyCollectionMock
+            ->method('count')
+            ->willReturn(0);
+
+        $this->sut = new tao_models_classes_dataBinding_GenerisInstanceDataBinder(
+            $this->target
+        );
+
+        $this->sut->withEventManager($this->eventManagerMock);
+    }
+
+    public function testBindScalarWithPreviousValue(): void
+    {
+        $this->eventManagerMock
+            ->expects($this->once())
+            ->method('trigger')
+            ->with($this->callback(function (MetadataModified $event): bool {
+                return (
+                    $event->getResource()->getUri() == $this->target->getUri()
+                    && $event->getMetadataUri() === self::URI_PROPERTY_1
+                    && $event->getMetadataValue() == 'Value 1');
+            }));
```

Comment:
I saw, that you repeat this one multiple times. Maybe you can extract this into a separate method and use it instead?
WDYT?

1. There's no need to wrap such condition with brackets
2. Use strict comparison
```suggestion
            ->with(
                $this->callback(
                    function (MetadataModified $event): bool {
                        return $event->getResource()->getUri() === $this->target->getUri()
                            && $event->getMetadataUri() === self::URI_PROPERTY_1
                            && $event->getMetadataValue() === 'Value 1';
                    }
                )
            );
```

Reply:
Good idea! I'm moving the setup for method expectations to a helper method in dd3faa491d1984a9fb7d729d62148922bd362d2f. Thank you :slightly_smiling_face: